src/braket/circuits/circuit.py

```diff
@@ -877,6 +882,77 @@ def to_ir(self) -> Program:
             basis_rotation_instructions=ir_basis_rotation_instructions,
         )
 
+    @staticmethod
+    def _einsum_subscripts(targets: QubitSet, qubit_count: int) -> str:
+        target_count = len(targets)
+
+        gate_left_indexes = list(range(target_count))
+        un_left_indexes = list(range(target_count, target_count + qubit_count))
+        un_right_indexes = list(range(target_count + qubit_count, target_count + 2 * qubit_count))
+
+        gate_right_indexes = [un_left_indexes[-1 - target] for target in targets]
+
+        result_left_indexes = un_left_indexes.copy()
+        for pos, target in enumerate(targets):
+            result_left_indexes[-1 - target] = gate_left_indexes[pos]
+
+        return (
+            gate_left_indexes + gate_right_indexes,
+            un_left_indexes + un_right_indexes,
+            result_left_indexes + un_right_indexes,
+        )
+
+    def as_unitary(self) -> np.ndarray:
+        """
+        Unitary matrix representation of the entire circuit.
+        Notice that any circuit with 10 qubits or more can get slow.
+
+        Returns:
+            np.ndarray: 2^qubit_count by 2^qubit_count circuit unitary matrix
+
```

Comment:
Minor doc suggestion:
```suggestion
        Returns the unitary matrix representation of the entire circuit.
        *Note*: The performance of this method degrades with qubit count. It might be slow for qubit count > 10.

        Returns:
            np.ndarray: A numpy array with shape (2^qubit_count, 2^qubit_count) representing the circuit as a unitary.
```

Might be helpful to have an example in the docstring as well. Refer to other docstrings in the circuit class.

Reply:
Done in https://github.com/aws/amazon-braket-sdk-python/pull/253/commits/afb28dd790933be8a9e4269db499a866ba0259d1